doc/user-manual/language/core-language.lagda.rst

```diff
@@ -8,21 +8,182 @@
 Core language
 *************
 
-.. note::
-   This is a stub
+A program in Agda consists of a number of declarations written in an ``*.agda``
+file. A declaration introduces a new identifier and gives its type and
+definition. It is possible to declare:
+
+* :ref:`datatypes <data-types>`
```

Comment:
Think on how to order these logically (or alt. alphabetically).

Reply:
I think logically would be better, is this correct?

* modules
* program options
* postulates
* variables
* datatypes
* record types
* pragmas
* function definitions
* local definitions
* pattern-synonyms
* precedence

Reply:
I could imagine going from frequent to exotic.
data, record, function, modules, local, postulates, variables, pattern-synonyms, precedence, pragmas, options